README.md

```diff
@@ -54,6 +63,403 @@ npm install @dojo/shim
+| type     | The `MatchType` that caused the outlet to render                        |
+| location | The location of the route that was matched                             |
+| params   | Key/Value object of the params that were parsed from the matched route |
+| router   | The router instance that can be used to provide functions that go to other routes/outlets|
+
+The following example uses `mapParams` to inject an `onClose` function that will go to the route registered against the `other-outlet` route and `id` property extracted from `params` in the `MyViewWidget` properties:
+
+```ts
+const FooOutlet = Outlet(MyViewWidget, 'foo', (options: MapParamsOptions) {
+	const { type, location, params, router } = options;
+
+	return {
+		onClose() {
+			// This creates a link for another outlet and set's the path
+			router.setPath(route.link('other-outlet'));
+		},
+		id: params.id
+	}
+});
+```
+
+##### Global Error Outlet
+
+Whenever a `MatchType.ERROR` occurs a global outlet is automatically added to the matched outlets called `errorOutlet`. This outlet can be used to render a widget for any unknown routes.
+
+```ts
+const ErrorOutlet = Outlet(ErrorWidget, 'errorOutlet');
+```
+
+#### Route Registration
+
+Routes are registered using `RouteConfig`, that defines a route's `path`, the associated `outlet` and nested child `RouteConfig`s. The full routes are recursively constructed from the nested route structure.
+
+Example routing configuration:
+
+```ts
+const config: RouteConfig[] = [
+	{
+		path: 'foo',
+		outlet: 'root',
+		children: [
+			{
+				path: 'bar',
+				outlet: 'bar'
+			},
+			{
+				path: 'baz',
+				children: [
+					{
+						path: 'qux'
+					}
+				]
+			}
+		]
+	}
+]
+```
+
+That would register the following routes and outlets:
+
+| Route        | Outlet |
+| ------------ | ------ |
+|`/foo`        | `root` |
+|`/foo/bar`    | `bar`  |
+|`/foo/baz`    | `baz`  |
+|`/foo/baz/qux`| `qux`  |
+
+**Note:** If an `outlet` is not explicitly specified the `path` will be used.
+
+To actually register the configuration with a Dojo 2 router, simply pass the config object in the `constructor` options
+
+```ts
+const router = new Router({ config });
+```
+
+For routes that have either path parameters or query parameters it is possible to specify default parameters. These parameters are used as a fallback when generating a link from an outlet without specifying parameters or the params do not exist in the current route.
+
+```ts
+const config = [
+	{
+		path: 'foo/{foo}',
+		outlet: 'foo',
+		defaultParams: {
+			foo: 'bar'
+		}
+	}
+];
+
+// Using router.link to generate a link for an outlet 'foo', will use the default 'bar' value
+router.link('foo')
+```
+
+A default route can be specified using the optional configuration property `defaultRoute` which will be used if the current route doesn't match an existing. Note there can only be one default route configured otherwise an error will be thrown.
```

Comment:
"`defaultRoute` which will be used if the current route doesn't match an existing." -> "`defaultRoute`, which will be used if the current route does not match an existing _______." (______ represents a missing word, presumably route?)